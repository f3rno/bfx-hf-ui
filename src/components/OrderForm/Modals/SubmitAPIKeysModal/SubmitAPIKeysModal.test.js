import React from 'react'
import { configure, mount, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import { Provider } from "react-redux"
import configureMockStore from "redux-mock-store"
import SubmitAPIKeysModal from './index'


const mockStore = configureMockStore()
const store = mockStore({
    exID: 'BFX'
})

configure({ adapter: new Adapter() })

describe('SubmitAPIKeysModal', () => {
    test('render modal', () => {
        const component = mount((
            <Provider store={store}>
                <SubmitAPIKeysModal 
                    exID='bitfinex' 
                />
            </Provider>
        ))

        expect(component).toMatchSnapshot()
    })
})

describe('SubmitAPIKeysModal', () => {
    test('require api credentials', () => {
        const component = mount((
            <Provider store={store}>
                <SubmitAPIKeysModal 
                    exID='bitfinex' 
                />
            </Provider>
        ))
        expect(component.find('.error').length).toBe(0)
        component.find('.green').simulate('click')
        expect(component.find('.error').length).toBe(1)
    })
})

describe('SubmitAPIKeysModal', () => {
    test('fill api credentials', () => {
        const component = mount((
                <SubmitAPIKeysModal 
                    exID='bitfinex' 
                    onSubmit={() => console.log('submited')}
                    onClose={() => console.log('closed')}
                />
        ))
        const submitBtn = component.find('.green')
        const apiKey = component.find('.hfui-settings__input-api-key').first()
        const apiSecret = component.find('.hfui-settings__input-api-secret').first()
        component.setState({apiKey: 'aaaa', apiSecret: '123'})
        submitBtn.simulate('click')
        console.log(component.state('error'))
        expect(component.find('.error').length).toBe(0)
    })
})