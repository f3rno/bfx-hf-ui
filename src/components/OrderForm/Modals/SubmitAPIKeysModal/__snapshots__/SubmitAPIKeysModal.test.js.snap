// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubmitAPIKeysModal render modal 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <SubmitAPIKeysModal
    exID="bitfinex"
  >
    <OrderFormModal
      buttons={
        <div
          className="row"
        >
          <Button
            className=""
            green={true}
            label="Submit"
            onClick={[Function]}
          />
          <Button
            className=""
            label="Cancel"
            red={true}
          />
        </div>
      }
      form={
        Array [
          <div
            className="row"
          >
            <Input
              className="hfui-settings__input-api-key"
              onChange={[Function]}
              placeholder="API KEY"
              type="text"
              value=""
            />
            <Input
              className="hfui-settings__input-api-secret"
              onChange={[Function]}
              placeholder="API SECRET"
              type="password"
              value=""
            />
          </div>,
          "",
        ]
      }
      icon="icon-api"
      isModal={true}
      title="SUBMIT API KEYS FOR BITFINEX"
    >
      <div
        className="hfui-orderform__modal-wrapper"
      >
        <Scrollbars>
          <Scrollbars
            autoHeight={false}
            autoHeightMax={200}
            autoHeightMin={0}
            autoHide={false}
            autoHideDuration={200}
            autoHideTimeout={1000}
            hideTracksWhenNotNeeded={false}
            renderThumbHorizontal={[Function]}
            renderThumbVertical={[Function]}
            renderTrackHorizontal={[Function]}
            renderTrackVertical={[Function]}
            renderView={[Function]}
            tagName="div"
            thumbMinSize={30}
            universal={false}
          >
            <div
              style={
                Object {
                  "height": "100%",
                  "overflow": "hidden",
                  "position": "relative",
                  "width": "100%",
                }
              }
            >
              <div
                key="view"
                style={
                  Object {
                    "WebkitOverflowScrolling": "touch",
                    "bottom": 0,
                    "left": 0,
                    "marginBottom": 0,
                    "marginRight": 0,
                    "overflow": "scroll",
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <div
                  className="hfui-orderform__modal-inner fullheight"
                  role="button"
                  tabIndex={0}
                >
                  <i
                    className="icon-api"
                  />
                  <p
                    style={Object {}}
                  >
                    SUBMIT API KEYS FOR BITFINEX
                  </p>
                  <div
                    className="hfui-orderform__modal-form"
                  >
                    <div
                      className="row"
                      key="form"
                    >
                      <Input
                        className="hfui-settings__input-api-key"
                        key="apiKey"
                        onChange={[Function]}
                        placeholder="API KEY"
                        type="text"
                        value=""
                      >
                        <div
                          className="hfui-input"
                        >
                          <input
                            className="hfui-settings__input-api-key"
                            onChange={[Function]}
                            placeholder="API KEY"
                            type="text"
                            value=""
                          />
                        </div>
                      </Input>
                      <Input
                        className="hfui-settings__input-api-secret"
                        key="apiSecret"
                        onChange={[Function]}
                        placeholder="API SECRET"
                        type="password"
                        value=""
                      >
                        <div
                          className="hfui-input"
                        >
                          <input
                            className="hfui-settings__input-api-secret"
                            onChange={[Function]}
                            placeholder="API SECRET"
                            type="password"
                            value=""
                          />
                          <button
                            className="field-icon"
                            onClick={[Function]}
                            type="button"
                          >
                            <Icon
                              Component="span"
                              name="eye"
                            >
                              <span
                                className="fa fa-eye"
                              />
                            </Icon>
                          </button>
                        </div>
                      </Input>
                    </div>
                  </div>
                  <div
                    className="hfui-orderform__modal-buttons"
                  >
                    <div
                      className="row"
                    >
                      <Button
                        className=""
                        green={true}
                        key="submit"
                        label="Submit"
                        onClick={[Function]}
                      >
                        <button
                          className="hfui-button green"
                          onClick={[Function]}
                          type="button"
                        >
                          Submit
                        </button>
                      </Button>
                      <Button
                        className=""
                        key="cancel"
                        label="Cancel"
                        red={true}
                      >
                        <button
                          className="hfui-button red"
                          type="button"
                        >
                          Cancel
                        </button>
                      </Button>
                    </div>
                  </div>
                </div>
              </div>
              <div
                key="trackHorizontal"
                style={
                  Object {
                    "borderRadius": 3,
                    "bottom": 2,
                    "display": "none",
                    "height": 6,
                    "left": 2,
                    "position": "absolute",
                    "right": 2,
                  }
                }
              >
                <div
                  style={
                    Object {
                      "backgroundColor": "rgba(0,0,0,.2)",
                      "borderRadius": "inherit",
                      "cursor": "pointer",
                      "display": "block",
                      "height": "100%",
                      "position": "relative",
                    }
                  }
                />
              </div>
              <div
                className="hfui-scrollbars-track-vertical"
                key="trackVertical"
                style={
                  Object {
                    "display": "none",
                    "position": "absolute",
                    "width": 6,
                  }
                }
              >
                <div
                  className="hfui-scrollbars-thumb-vertical"
                  style={
                    Object {
                      "display": "block",
                      "position": "relative",
                      "width": "100%",
                    }
                  }
                />
              </div>
            </div>
          </Scrollbars>
        </Scrollbars>
      </div>
    </OrderFormModal>
  </SubmitAPIKeysModal>
</Provider>
`;
